lib/Zonemaster/Backend/DB.pm

```diff
@@ -105,23 +105,14 @@ sub dbh {
     return $self->dbhandle;
 }
 
-sub user_exists {
-    my ( $self, $user ) = @_;
-
-    die Zonemaster::Backend::Error::Internal->new( reason => "username not provided to the method user_exists")
-        unless ( $user );
-
-    return $self->user_exists_in_db( $user );
-}
-
 sub add_api_user {
     my ( $self, $username, $api_key ) = @_;
 
     die Zonemaster::Backend::Error::Internal->new( reason => "username or api_key not provided to the method add_api_user")
         unless ( $username && $api_key );
 
     die Zonemaster::Backend::Error::Conflict->new( message => 'User already exists', data => { username => $username } )
-        if ( $self->user_exists( $username ) );
+        if ( $self->user_exists_in_db( $username ) );
```

Comment:
Have you considered taking this cleanup one step further? It looks to me like add_api_user only ever throws "add_api_user_to_db not successful" if there is a race between user_exists_in_db() and add_api_user_to_db(). If this is true we could simply remove the user_exists_in_db() check and have the $result check throw 'User already exists' instead. WDYT?

Reply:
I just took for granted that it was already unique.. I'm fine with a separate PR.